src/scenario.py

```diff
@@ -425,7 +453,9 @@ def run(self, strategy_name, options):
             if ext != "csv":
                 print("File extension mismatch: timeseries file is of type .csv")
 
-            cs_ids = sorted(strat.world_state.charging_stations.keys())
+            cs_ids = sorted(item for item in strat.world_state.charging_stations.keys() if
+                            self.constants.charging_stations[item].parent == gcID)
```

Comment:
what is gcID? I think it is the last GC looked at after the simulation. You'd need to have an array for each GC -> `cs_ids = {gcID: sorted(item ... if cs[item].parent == gcID) for gcID in gc_ids). This also goes for a lot of the lines after this. You have to iterate over the GCs, otherwise you only get one! Is it really your intention to create output files for each GC individually?

Reply:
yes, each gc has its own output file